packages/package-requester/src/packageRequester.ts

```diff
@@ -376,6 +376,9 @@ function fetchToStore (
   ) {
     try {
       const isLocalTarballDep = opts.pkgId.startsWith('file:')
+      const pkgIndexFilePath = opts.resolution['integrity']
```

Comment:
- This exact code is used on line 45 of packages/plugin-commands-store/src/storeStatus/index.ts. It would be good to extract it into a separate function somewhere.

- Since this is TypeScript, it would be nice to use an enum for the mode value.

Reply:
For now, I am leaving it as it is because the store structure will have more changes and I don't know where to put it now.